File: php_scripts_examples/decoder.py
# decode_response.py
import json
import base64

# --- Configuration ---
# The name of the JSON file  saved from  curl command
INPUT_JSON_FILE = "decoder/response_pdf.json"

# --- Script ---
print(f"Reading response from '{INPUT_JSON_FILE}'...")

try:
    with open(INPUT_JSON_FILE, 'r') as f:
        data = json.load(f)
except FileNotFoundError:
    print(f"Error: The file '{INPUT_JSON_FILE}' was not found.")
    print("Please make sure you've run the curl command and saved the output correctly.")
    exit()
except json.JSONDecodeError:
    print(f"Error: The file '{INPUT_JSON_FILE}' is not a valid JSON file.")
    exit()

# Decode the non-anonymous report
non_anon_base64 = data.get("non_anonymous_report_base64")
if non_anon_base64:
    non_anon_bytes = base64.b64decode(non_anon_base64)
    output_filename_non_anon = "DECODED_non_anonymous_report.pdf"
    with open(output_filename_non_anon, 'wb') as f_out:
        f_out.write(non_anon_bytes)
    print(f"✅ Successfully saved '{output_filename_non_anon}'")
else:
    print("⚠️  'non_anonymous_report_base64' key not found in JSON.")

# Decode the anonymous report
anon_base64 = data.get("anonymous_report_base64")
if anon_base64:
    anon_bytes = base64.b64decode(anon_base64)
    output_filename_anon = "DECODED_anonymous_report.pdf"
    with open(output_filename_anon, 'wb') as f_out:
        f_out.write(anon_bytes)
    print(f"✅ Successfully saved '{output_filename_anon}'")
else:
    print("⚠️  'anonymous_report_base64' key not found in JSON.")

File: php_scripts_examples/decoder.php
<?php

// decode_response.php

// --- Configuration ---
$inputJsonFile = 'response_pdf.json';

echo "Reading response from '{$inputJsonFile}'...\n";

// --- Script ---

// 1. Check if the file exists and is readable
if (!is_readable($inputJsonFile)) {
    die("Error: The file '{$inputJsonFile}' was not found or cannot be read.\n");
}

// 2. Read the file content into a string
$jsonContent = file_get_contents($inputJsonFile);

// 3. Decode the JSON string into an associative array
// The 'true' argument converts JSON objects to associative arrays
$data = json_decode($jsonContent, true);

// Check if JSON decoding was successful
if (json_last_error() !== JSON_ERROR_NONE) {
    die("Error: Invalid JSON format in '{$inputJsonFile}'.\n");
}

// 4. Decode and save the non-anonymous report
if (isset($data['non_anonymous_report_base64'])) {
    // Get the Base64 string from the array
    $nonAnonBase64 = $data['non_anonymous_report_base64'];

    // Decode the string into raw binary data
    $nonAnonBytes = base64_decode($nonAnonBase64);

    // Save the binary data to a new PDF file
    $outputFilenameNonAnon = 'DECODED_non_anonymous_report.php.pdf';
    file_put_contents($outputFilenameNonAnon, $nonAnonBytes);

    echo "✅ Successfully saved '{$outputFilenameNonAnon}'\n";
} else {
    echo "⚠️  'non_anonymous_report_base64' key not found in JSON.\n";
}

// 5. Decode and save the anonymous report
if (isset($data['anonymous_report_base64'])) {
    // Get the Base64 string from the array
    $anonBase64 = $data['anonymous_report_base64'];

    // Decode the string
    $anonBytes = base64_decode($anonBase64);

    // Save the file
    $outputFilenameAnon = 'DECODED_anonymous_report.php.pdf';
    file_put_contents($outputFilenameAnon, $anonBytes);

    echo "✅ Successfully saved '{$outputFilenameAnon}'\n";
} else {
    echo "⚠️  'anonymous_report_base64' key not found in JSON.\n";
}

?>